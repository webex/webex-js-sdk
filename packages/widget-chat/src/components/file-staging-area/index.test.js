/* eslint max-nested-callbacks: ["error", 3] */
import React from 'react';
import renderer from 'react-test-renderer';
import {Map} from 'immutable';

import fh from '@ciscospark/test-helper-file';

import FileStagingArea from '.';

describe(`FileStagingArea component`, () => {
  let component, props;
  let txtFile = `sample-text-one.txt`;
  let pptFile = `sample-powerpoint-two-page.ppt`;

  beforeAll(() =>
    Promise.all([
      fh.fetchWithoutMagic(txtFile),
      fh.fetchWithoutMagic(pptFile)
    ])
    .then((res) => {
      [txtFile, pptFile] = res;
      txtFile.type = `text/plain`;
      pptFile.type = `application/vnd.openxmlformats-officedocument.presentationml.presentation`;
    })
  );

  beforeEach(() => {
    props = {
      files: new Map({txtFile, pptFile}),
      onFileRemove: jest.fn(),
      onSubmit: jest.fn()
    };

    component = renderer.create(
      <FileStagingArea {...props} />
    );
  });


  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
