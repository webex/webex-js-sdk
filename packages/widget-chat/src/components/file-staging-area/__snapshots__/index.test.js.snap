exports[`FileStagingArea component renders properly 1`] = `
<div
  className="file-staging-area">
  <div
    className="staged-files">
    <div
      className="chip-container">
      <div
        className="chip">
        <div
          className="file-thumbnail">
          <div
            className="file-icon">
            <div
              className="icon"
              title={undefined} />
          </div>
          <img
            role="presentation"
            src={undefined} />
        </div>
        <div
          className="file-info">
          <div
            className="file-name">
            sample-text-one.txt
          </div>
          <div
            className="file-meta">
            <div
              className="file-size">
              0 Bytes
            </div>
            <div
              className="file-type">
              Plain file
            </div>
          </div>
        </div>
        <div
          className="chip-action">
          <button
            className="button"
            onClick={[Function]}
            onKeyUp={[Function]}>
            <div
              className="icon"
              title={undefined} />
          </button>
        </div>
      </div>
    </div>
    <div
      className="chip-container">
      <div
        className="chip">
        <div
          className="file-thumbnail">
          <div
            className="file-icon">
            <div
              className="icon"
              title={undefined} />
          </div>
          <img
            role="presentation"
            src={undefined} />
        </div>
        <div
          className="file-info">
          <div
            className="file-name">
            sample-powerpoint-two-page.ppt
          </div>
          <div
            className="file-meta">
            <div
              className="file-size">
              0 Bytes
            </div>
            <div
              className="file-type">
              presentation
            </div>
          </div>
        </div>
        <div
          className="chip-action">
          <button
            className="button"
            onClick={[Function]}
            onKeyUp={[Function]}>
            <div
              className="icon"
              title={undefined} />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="staging-actions">
    <button
      className="button"
      onClick={[Function]}
      onKeyUp={[Function]}>
      Share
    </button>
  </div>
</div>
`;
